### PR TITLE
[Feature] Add p0f scanner

### PR DESCRIPTION
**Pull request by denpamusic**
As per discussion in #968, this PR adds support for passive OS fingerprinting (p0f) as part of lua_scanners framework. It also includes functional tests and optional header routine in milter headers module. Redis cache will be used by this plugin if available but otherwise not required.

### CODE REVIEW

**Comment by denpamusic** [comment]
Sorry forgot to chmod +x dummy_p0f...

**Comment by denpamusic** [comment]
Cleaned up config a bit and added test case for working without redis just in case. This should be final commit. Sorry for the inconvenience.

**Comment by vstakhov** [comment]
That's really great work! Thank you, Denis!

**Comment by denpamusic** [comment]
You're welcome! Always glad to learn something new and help great project in process.

File: conf/modules.d/p0f.conf [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# '$LOCAL_CONFDIR/override.d/file.conf' - to override the defaults
+#
+# See https://rspamd.com/doc/tutorials/writing_rules.html for details
+
+p0f {
+  # Disable module by default
+  enabled = false;
+
+  # Path to the unix socket that p0f listens on
+  socket = '/tmp/p0f.sock';
```
Comment (moisseev) [inline conf/modules.d/p0f.conf]: @denpamusic Would you explain the reason of using `/tmp` for unix sockets? Wouldn't it be better to use `/var/run/p0f.sock` instead?

Comment (denpamusic) [reply conf/modules.d/p0f.conf]: Sorry, I've come from FreeBSD-land where most sockets created in /tmp by default as /var/run in only writable by root, so it's kinda habit of mine... You're welcome to change it of course, if you think that I'll be better suited for most users. This PR is already merged, so you'll need to create new one.

Comment (denpamusic) [reply conf/modules.d/p0f.conf]: After doing a bit of research I've decided to change default socket location to `/var/run/p0f.sock` as you've recommended. I've submitted #3043 that includes this change as well as fixes a couple other problems I've noticed during plugin usage.

Comment (moisseev) [reply conf/modules.d/p0f.conf]: Are you running p0f as an unprivileged user? How? AFAIU it needs permission to read from `BPF`.

Comment (denpamusic) [reply conf/modules.d/p0f.conf]: You're right, but p0f can deal with it with `-u user` flag. Basically it's opens bpf as root and the drops privileges.

**Pull request merged**